### PR TITLE
[BugFix][Kernel]: fix illegal memory access in causal_conv1d when conv_states is None

### LINKED ISSUE #10895: [Bug]: illegal memory access in `causal_conv1d_fn` with input length 1026
### Your current environment

<details>
<summary>The output of `python collect_env.py`</summary>

```text
PyTorch version: 2.5.1+cu124
Is debug build: False
CUDA used to build PyTorch: 12.4
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.4 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version: Could not collect
CMake version: version 3.28.3
Libc version: glibc-2.35

Python version: 3.10.15 (main, Oct  3 2024, 07:27:34) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.15.0-91-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: 12.4.99
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: 
GPU 0: NVIDIA H100 80GB HBM3
GPU 1: NVIDIA H100 80GB HBM3

Nvidia driver version: 535.161.08
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.9.0.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv.so.9.0.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn.so.9.0.0
/usr/lib/x86_64-linux-gnu/libcudnn_engines_precompiled.so.9.0.0
/usr/lib/x86_64-linux-gnu/libcudnn_engines_runtime_compiled.so.9.0.0
/usr/lib/x86_64-linux-gnu/libcudnn_graph.so.9.0.0
/usr/lib/x86_64-linux-gnu/libcudnn_heuristic.so.9.0.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops.so.9.0.0
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Address sizes:                      46 bits physical, 57 bits virtual
Byte Order:                         Little Endian
CPU(s):                             192
On-line CPU(s) list:                0-191
Vendor ID:                          GenuineIntel
Model name:                         Intel(R) Xeon(R) Platinum 8468
CPU family:                         6
Model:                              143
Thread(s) per core:                 2
Core(s) per socket:                 48
Socket(s):                          2
Stepping:                           8
BogoMIPS:                           4200.00
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cat_l2 cdp_l3 invpcid_single intel_ppin cdp_l2 ssbd mba ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect avx_vnni avx512_bf16 wbnoinvd dtherm arat pln pts avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid bus_lock_detect cldemote movdiri movdir64b enqcmd fsrm md_clear serialize tsxldtrk pconfig arch_lbr amx_bf16 avx512_fp16 amx_tile amx_int8 flush_l1d arch_capabilities
L1d cache:                          4.5 MiB (96 instances)
L1i cache:                          3 MiB (96 instances)
L2 cache:                           192 MiB (96 instances)
L3 cache:                           210 MiB (2 instances)
NUMA node(s):                       2
NUMA node0 CPU(s):                  0-47,96-143
NUMA node1 CPU(s):                  48-95,144-191
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Not affected
Vulnerability Retbleed:             Not affected
Vulnerability Spec rstack overflow: Not affected
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Enhanced IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected

Versions of relevant libraries:
[pip3] flake8==7.1.1
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.26.4
[pip3] nvidia-cublas-cu12==12.4.5.8
[pip3] nvidia-cuda-cupti-cu12==12.4.127
[pip3] nvidia-cuda-nvrtc-cu12==12.4.127
[pip3] nvidia-cuda-runtime-cu12==12.4.127
[pip3] nvidia-cudnn-cu12==9.1.0.70
[pip3] nvidia-cufft-cu12==11.2.1.3
[pip3] nvidia-curand-cu12==10.3.5.147
[pip3] nvidia-cusolver-cu12==11.6.1.9
[pip3] nvidia-cusparse-cu12==12.3.1.170
[pip3] nvidia-ml-py==12.560.30
[pip3] nvidia-nccl-cu12==2.21.5
[pip3] nvidia-nvjitlink-cu12==12.4.127
[pip3] nvidia-nvtx-cu12==12.4.127
[pip3] pyzmq==26.2.0
[pip3] torch==2.5.1
[pip3] torchvision==0.20.1
[pip3] transformers==4.46.3
[pip3] triton==3.1.0
[conda] numpy                     1.26.4                   pypi_0    pypi
[conda] nvidia-cublas-cu12        12.4.5.8                 pypi_0    pypi
[conda] nvidia-cuda-cupti-cu12    12.4.127                 pypi_0    pypi
[conda] nvidia-cuda-nvrtc-cu12    12.4.127                 pypi_0    pypi
[conda] nvidia-cuda-runtime-cu12  12.4.127                 pypi_0    pypi
[conda] nvidia-cudnn-cu12         9.1.0.70                 pypi_0    pypi
[conda] nvidia-cufft-cu12         11.2.1.3                 pypi_0    pypi
[conda] nvidia-curand-cu12        10.3.5.147               pypi_0    pypi
[conda] nvidia-cusolver-cu12      11.6.1.9                 pypi_0    pypi
[conda] nvidia-cusparse-cu12      12.3.1.170               pypi_0    pypi
[conda] nvidia-ml-py              12.560.30                pypi_0    pypi
[conda] nvidia-nccl-cu12          2.21.5                   pypi_0    pypi
[conda] nvidia-nvjitlink-cu12     12.4.127                 pypi_0    pypi
[conda] nvidia-nvtx-cu12          12.4.127                 pypi_0    pypi
[conda] pyzmq                     26.2.0                   pypi_0    pypi
[conda] torch                     2.5.1                    pypi_0    pypi
[conda] torchvision               0.20.1                   pypi_0    pypi
[conda] transformers              4.46.3                   pypi_0    pypi
[conda] triton                    3.1.0                    pypi_0    pypi
ROCM Version: Could not collect
Neuron SDK Version: N/A
vLLM Version: 0.6.4.post1
vLLM Build Flags:
CUDA Archs: 5.2 6.0 6.1 7.0 7.2 7.5 8.0 8.6 8.7 9.0+PTX; ROCm: Disabled; Neuron: Disabled
GPU Topology:
GPU0    GPU1    NIC0    NIC1    NIC2    NIC3    NIC4    NIC5    NIC6    NIC7    NIC8    NIC9    NIC10   NIC11   CPU Affinity    NUMA Affinity   GPU NUMA ID
GPU0     X      NV18    SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     PIX     NODE    NODE    NODE    48-95,144-191  1N/A
GPU1    NV18     X      SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     NODE    NODE    PIX     NODE    48-95,144-191  1N/A
NIC0    SYS     SYS      X      NODE    NODE    NODE    NODE    NODE    NODE    NODE    SYS     SYS     SYS     SYS
NIC1    SYS     SYS     NODE     X      NODE    NODE    NODE    NODE    NODE    NODE    SYS     SYS     SYS     SYS
NIC2    SYS     SYS     NODE    NODE     X      NODE    NODE    NODE    NODE    NODE    SYS     SYS     SYS     SYS
NIC3    SYS     SYS     NODE    NODE    NODE     X      PIX     PXB     PXB     NODE    SYS     SYS     SYS     SYS
NIC4    SYS     SYS     NODE    NODE    NODE    PIX      X      PXB     PXB     NODE    SYS     SYS     SYS     SYS
NIC5    SYS     SYS     NODE    NODE    NODE    PXB     PXB      X      PIX     NODE    SYS     SYS     SYS     SYS
NIC6    SYS     SYS     NODE    NODE    NODE    PXB     PXB     PIX      X      NODE    SYS     SYS     SYS     SYS
NIC7    SYS     SYS     NODE    NODE    NODE    NODE    NODE    NODE    NODE     X      SYS     SYS     SYS     SYS
NIC8    PIX     NODE    SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS      X      NODE    NODE    NODE
NIC9    NODE    NODE    SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     NODE     X      NODE    NODE
NIC10   NODE    PIX     SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     NODE    NODE     X      NODE
NIC11   NODE    NODE    SYS     SYS     SYS     SYS     SYS     SYS     SYS     SYS     NODE    NODE    NODE     X 

Legend:

  X    = Self
  SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
  NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
  PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
  PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
  PIX  = Connection traversing at most a single PCIe bridge
  NV#  = Connection traversing a bonded set of # NVLinks

NIC Legend:

  NIC0: mlx5_0
  NIC1: mlx5_1
  NIC2: mlx5_2
  NIC3: mlx5_3
  NIC4: mlx5_4
  NIC5: mlx5_5
  NIC6: mlx5_6
  NIC7: mlx5_7
  NIC8: mlx5_8
  NIC9: mlx5_9
  NIC10: mlx5_10
  NIC11: mlx5_11

CUDA_CACHE_DISABLE=1
CUBLAS_VERSION=12.4.2.65
NVIDIA_DRIVER_CAPABILITIES=compute,utility,video
NCCL_IB_RETRY_CNT=12
NCCL_IB_QPS_PER_CONNECTION=8
NVIDIA_BUILD_ID=85286408
NCCL_IB_TIMEOUT=22
TORCH_CUDA_ARCH_LIST=5.2 6.0 6.1 7.0 7.2 7.5 8.0 8.6 8.7 9.0+PTX
NCCL_IB_DISABLE=0
NCCL_IB_HCA=mlx5_0,mlx5_1,mlx5_2,mlx5_7,mlx5_8,mlx5_9,mlx5_10,mlx5_11
NCCL_DEBUG=INFO
NCCL_WORK_FIFO_DEPTH=4194304
CUDA_HOME=/usr/local/cuda
CUDA_HOME=/usr/local/cuda
PYTORCH_BUILD_NUMBER=0
NCCL_VERSION=2.20.5
NCCL_SOCKET_IFNAME=eth0
CUDA_PATH=/usr/local/cuda
PYTORCH_HOME=/opt/pytorch/pytorch
NVIDIA_PRODUCT_NAME=PyTorch
CUDA_DRIVER_VERSION=550.54.14
LD_LIBRARY_PATH=/root/miniconda3/envs/mamba/lib/python3.10/site-packages/cv2/../../lib64:/usr/local/cuda/lib64:/usr/local/lib/python3.10/dist-packages/torch/lib:/usr/local/lib/python3.10/dist-packages/torch_tensorrt/lib:/usr/local/cuda/compat/lib:/usr/local/nvidia/lib:/usr/local/nvidia/lib64
PYTORCH_BUILD_VERSION=2.3.0a0+40ec155e58
CUDNN_VERSION=9.0.0.306+cuda12.3
NVIDIA_REQUIRE_CUDA=cuda>=9.0
NVIDIA_PYTORCH_VERSION=24.03
CUDA_VERSION=12.4.0.041
TORCH_ALLOW_TF32_CUBLAS_OVERRIDE=1
CUDACXX=/usr/local/cuda/bin/nvcc
NVIDIA_REQUIRE_JETPACK_HOST_MOUNTS=
MAX_JOBS=8
CUDA_MODULE_LOADING=LAZY
PYTORCH_VERSION=2.3.0a0+40ec155e58
NVIDIA_VISIBLE_DEVICES=GPU-5d4ee582-efa3-e4d2-4363-c7a533ffc3cb,GPU-421dbc96-2290-09cd-acfa-8ebfa5554ab0
NCCL_NVLS_ENABLE=0
NCCL_SOCKET_FAMILY=AF_INET
```

</details>

### Model Input Dumps

_No response_

### 🐛 Describe the bug

I encountered an **illegal memory access** error when using the `causal_conv1d_fn` function with an input tensor shape of [1, dim, 1026]. This issue does not occur with other sequence lengths like 1024 or 1028.

### Reproduction code
```python
from vllm.model_executor.layers.mamba.ops.causal_conv1d import causal_conv1d_fn
import torch
# from causal_conv1d import causal_conv1d_fn

dim = 1024
seq_len = 1026

x = (
    torch.randn(1, dim, seq_len).to(torch.bfloat16).cuda()
)

weight = (
    torch.randn(dim, 4).to(torch.bfloat16).cuda()
)

activation = "silu"

output = causal_conv1d_fn(x, weight, None, activation=activation)

print(output)
```

### Observation

The illegal memory access error did not occur when I used the ·causal_conv1d· function from the [causal_conv1d](https://github.com/Dao-AILab/causal-conv1d) repository instead of causal_conv1d_fn.

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by xffxff**
fix #10895 

Issue #10895 reports an illegal memory access error on H100 GPUs when calling the `causal_conv1d_fn` function with specific input shapes and when `conv_states` is not provided. After some investigation, I found that the root cause as an attempt to assign a value to a `nullptr`. 

cc @mzusman @tlrmchlsmth

### CODE REVIEW

**Comment by mzusman** [comment]
Thanks for fixing the bug! it would be great if you could add a test case to [test_causal_conv1d.py](https://github.com/xffxff/vllm/blob/ba25e7b1b162596e7d49891e57fe94b548f0fb96/tests/kernels/test_causal_conv1d.py#L157) for running the causal_conv1d without initial state

**Comment by xffxff** [comment]
> it would be great if you could add a test case to [test_causal_conv1d.py](https://github.com/xffxff/vllm/blob/ba25e7b1b162596e7d49891e57fe94b548f0fb96/tests/kernels/test_causal_conv1d.py#L157) for running the causal_conv1d_fn kernel without an initial state

@mzusman I've added tests for `causal_conv1d_fn` kernel without an initial state and all the tests passed successfully in my local environment.

**Comment by mzusman** [comment]
> > it would be great if you could add a test case to [test_causal_conv1d.py](https://github.com/xffxff/vllm/blob/ba25e7b1b162596e7d49891e57fe94b548f0fb96/tests/kernels/test_causal_conv1d.py#L157) for running the causal_conv1d_fn kernel without an initial state
> 
> @mzusman I've added tests for `causal_conv1d_fn` kernel without an initial state and all the tests passed successfully in my local environment.

Thanks! LGTM @tlrmchlsmth

**Review by mzusman — APPROVED** [review]

**Review by jeejeelee — APPROVED** [review]
Thank you very much. LGTM, leaving the final decision to @tlrmchlsmth

**Review by tlrmchlsmth — APPROVED** [review]
Thanks for the contribution!
